programs/pkey/dh_client.c

```diff
@@ -30,18 +29,19 @@
 
 #define SERVER_NAME "localhost"
 #define SERVER_PORT "11999"
+#define MBEDTLS_MD_CAN_SHA256_MAX_SIZE 32
```

Comment:
This one is tricky. We cannot use `MBEDTLS_MD_MAX_SIZE` because when the rsa header is imported it will be set to the maximum possible size of 64 because of how md.h is setting it to the [maximum ](https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/md.h#L70-L81)size. 

That works when we need to allocate a buffer that can fit any possible md, but the un-ininitialised memory can cause bugs, if the mbedtls_shaX() method only modifies an N ammount of bytes which is less than `MBEDTLS_MD_MAX_SIZE`

Users should be advised to either:
- calloc() the buffer allocated by `MBEDTLS_MD_MAX_SIZE`
- bound the buffer they are using to the output of the `mbedtls_shaX()` and make certain that that consistently used in our API calls.

Reply:
This isn't really `MBEDTLS_MD_CAN_SHA256_MAX_SIZE`, it's really just `SHA256_SIZE`